community/rfcs/25-03-07-OPEA-001-OIM-Operator.md

```diff
@@ -0,0 +1,57 @@
+
+## Motivation
+
+Nvidia delivers this ease of use through tooling that first detects the runtime hardware, which is then used to filter for profiles given the desired model and KPIs. A profile is a recipe specifying how to launch an inference service.  It specifies not only which model server framework (such as TGI or vLLM or Ollama ) to use but also number GPU cards, amount of memory, degree of tensor parallelism, batch size etc. Lastly the profile is used to launch the inference service on the given hardware. 
+
+The profiles are established through offline experimentation using tools like MLPerf. For a model, one or more model frameworks are explored on different hardware to determine settings the deliver on the desired KPIs. Constructing profiles is non-trivial, resource intensive work.  
+
+## Design Proposal
+
+We seek to offer a similar ease of use experience in OPEA for the inference services. Nvidia’s open source NIM Operator is a tool that eases launching inference services. OPEA plans to extend it to deliver an Operator that provides similar functionality but also supports multi-vendor hardware and shall endeavor to work with Kubernetes and flavors such as Red Hat OpenShift and SUSE Rancher, their AI variants by working with our partners and potentially integrating with platforms such as KServe and RayAI.  The first release will provide minimal functionality and over time with community contributions grow richer.  We anticipate providers of AI platforms that operate on a multitude of hardware platforms will find the HW optimized profiles valuable in providing cost-effective performant inference services.
+
+With regard to profiles we shall start with supporting vLLM as the model framework and build profiles for a few popular models. We do anticipate constant enhancements and new offerings in the model serving space and a steady influx of new models.
+
+The work will entail 
+
+1) auto-detecting hardware and what is allocated
+
+2) Taking the user specified model and profile (if not specified using a default profile) to retrieve configurations that are possible on the allocated hardware type and quantity. For instance, a configuration that leverages tensor parallelism is infeasible if only one GPU card is allocated. 
+
+3) Automating performance studies to identify optimum settings for a given hardware with various resource allocations for a given model and constructing profile specific configurations. These studies will be repeated to take advantage of new models, frameworks and other enhancements.  
+
+4) Providing a tool to discover and list available profiles/configurations for a model. Using this tool in the OPEA Operator, and within our inference microservices to identify relevant configurations and use. 
+
```

Comment:
Could 4 & 5 be merged, as in use ORAs from within the Operator, container ..?

Reply:
done.